File: backend/src/main/java/com/efedotov/vaultly/dto/file/LinkFileRequest.java
package com.efedotov.vaultly.dto.file;

import jakarta.validation.constraints.Size;
import lombok.Data;

import java.util.UUID;

@Data
public class LinkFileRequest {
    private UUID fileContentId;

    @Size(max = 500)
    private String fileName;

    private UUID folderId;
    private Boolean isPublic;
}

File: lib/server/model/file/link_file_request.dart
class LinkFileRequest {
  final String? fileContentId;
  final String? fileName;
  final String? folderId;
  final bool? isPublic;

  LinkFileRequest({
    this.fileContentId,
    this.fileName,
    this.folderId,
    this.isPublic,
  });

  factory LinkFileRequest.fromJson(Map<String, dynamic> json) {
    return LinkFileRequest(
      fileContentId: json['fileContentId'] as String?,
      fileName: json['fileName'] as String?,
      folderId: json['folderId'] as String?,
      isPublic: json['isPublic'] as bool?,
    );
  }

  Map<String, dynamic> toJson() => {
    if (fileContentId != null) 'fileContentId': fileContentId,
    if (fileName != null) 'fileName': fileName,
    if (folderId != null) 'folderId': folderId,
    if (isPublic != null) 'isPublic': isPublic,
  };
}
